index.bs

```diff
@@ -657,132 +882,160 @@ dictionary FilePropertyBag : BlobPropertyBag {
+1. If reading bytes failed other than by reaching the end of the file, throw a {{NotReadableError}}.
+1. If reading bytes failed becaus the end of the file was reached, return [=end of blob=].
+1. Set |read state|.[=file blob read state/offset=] to |read state|.[=file blob read state/offset=] + |bytes|'s [=byte sequence/length=].
+1. Return |bytes|.
 
-4. Initialize the value of |value|'s {{File/lastModified}} attribute to
-   |serialized|.\[[LastModified]].
+</div>
 
 ## Constructor ## {#file-constructor}
 
 <div algorithm="file-constructor">
-The {{File}} constructor is invoked with two or three parameters,
-depending on whether the optional dictionary parameter is used.
-When the {{File()}} constructor is invoked,
-user agents must run the following steps:
-
-1. Let |bytes| be the result of [=processing blob parts=] given {{fileBits}}
-  and {{File/File(fileBits, fileName, options)/options}}.
+The <dfn constructor for=File lt="File(fileBits, fileName, options)|File(fileBits, fileName)"><code>new File(|fileBits|, |fileName|, |options|)</code></dfn> constructor steps are:
 
+1. Run the {{Blob(blobParts, options)|Blob}}(|fileBits|, |options|) constructor steps.
 
-2. Let |n| be a new string of the same size as the {{fileName}} argument to the constructor.
-  Copy every character from {{fileName}} to |n|,
+1. Let |n| be a new string of the same size as |fileName|.
+1. Copy every character from |fileName| to |n|,
   replacing any "/" character (U+002F SOLIDUS) with a ":" (U+003A COLON).
 
   Note: Underlying OS filesystems use differing conventions for file name;
-  with constructed files, mandating UTF-16 lessens ambiquity when file names are converted to <a>byte</a> sequences.
-
-3. Process {{FilePropertyBag}} dictionary argument by running the following substeps:
-
-  1. If the {{BlobPropertyBag/type}} member is provided and is not the empty string,
-    let |t| be set to the {{BlobPropertyBag/type}} dictionary member.
-    If |t| contains any characters outside the range U+0020 to U+007E,
-    then set |t| to the empty string and return from these substeps.
-  2. Convert every character in |t| to [=ASCII lowercase=].
-  3. If the {{FilePropertyBag/lastModified}} member is provided,
-    let |d| be set to the  {{FilePropertyBag/lastModified}} dictionary member.
-    If it is not provided,
-    set |d| to the current date and time
+  with constructed files, mandating UTF-16 lessens ambiquity when file names are converted to <a>byte sequences</a>.
+
+1. If |options|.{{FilePropertyBag/lastModified}} member is provided:
+  1. Let |d| be |options|.{{FilePropertyBag/lastModified}} dictionary member.
+1. Otherwise:
+  1. Let |d| be the current date and time
     represented as the number of milliseconds since the <a>Unix Epoch</a>
     (which is the equivalent of <code>Date.now()</code> [[ECMA-262]]).
 
     Note: Since ECMA-262 {{Date}} objects convert to <code>long long</code> values
```

Comment:
Maybe move this into the domintro ?

Reply:
I guess File constructor doesn't have a domintro yet. Add one? Blob too?

That'd be a good place to call out (in giant blinking text...) that the first argument is a array. :)